Sources/BrowserServicesKit/ContentBlocking/AdClickAttribution/AdClickAttributionLogic.swift

```diff
@@ -97,6 +97,7 @@ public class AdClickAttributionLogic {
         case .noAttribution:
             self.state = state
         case .preparingAttribution(let vendor, let info, _):
+            cancelAttributedRulesCompilationIfNeeded()
```

Comment:
`applyInheritedAttribution` should be called only when creating new tab from the context of the attributed tab - it should have no existing state to "cancel" at that point, so if it has, this feels like a possible problem.